spatial-deathball-unity/Assets/Scripts/BallControl.cs

```diff
@@ -46,59 +46,170 @@ private void FixedUpdate()
         }
     }
 
-    private bool TryFindNewTarget(bool resetVelocity)
+    private bool TryFindNewTarget(bool resetVelocity, bool targetClosest = false)

```

Comment:
> Deterministically choosing what player the Bot/Ball Targets

Theres a few ways you could go about this,
1. Currently the owner of the ball is fully in control of it for the whole game. When a bot blocks a ball it could takeover ownership of the ball and dictate the new target from there.
>This was actually my original plan. Each time the ball targets `you` or a local `bot`, you takeover the ball so that you sim it locally. Remove any network lag etc, but I felt it made the code a little messy and obviously vulnerable so for now we have single ownership on the ball.
2. A bot could pre-decide if it will block and who it will target and sync this state. Then when the ball hits the bot the master-client it will read the state and make it happen.
3. You can message between networked clients with RPCs using our [networking service](https://cs.spatial.io/reference/SpatialSys.UnitySDK.INetworkingRemoteEventsService).


`2` is probably this simplest to add to the current code.

Reply:
Thanks Alan! I'll think about this and see what changes can be made 👍